auth/routes.py

```diff
@@ -0,0 +1,95 @@
+from fastapi import APIRouter, Depends, HTTPException, status
+from fastapi.security import OAuth2PasswordBearer, OAuth2PasswordRequestForm
+from sqlalchemy.orm import Session
+from auth.schemas import CreateUsers, Token, TokenData
+from auth.utils import verify_password, get_password_hash
+from auth.models import Users
+from database.database import get_db 
+from jose import jwt, JWTError
+from datetime import timedelta 
+
+# Import these from wherever they are defined
+from auth.utils import SECRET_KEY, ALGORITHM, ACCESS_TOKEN_EXPIRE_MINUTES
+
+oauth_2_scheme = OAuth2PasswordBearer(tokenUrl="token")
+router = APIRouter()
+
+    
+def authenticate_user(db, username: str, password: str):
+
+    user = db.query(Users).filter(Users.username == username).first() 
```

Comment:
Instead of accessing db directly in controllers. We should create a data access layer. So for this example User related queries should be in a user service module. Research on pro and cons of a data access layer.